pkg/driver/aws-efs/aws_efs_tags_controller.go

```diff
@@ -0,0 +1,413 @@
+		infra.Status.PlatformStatus.AWS.ResourceTags != nil {
+		err := c.fetchPVsAndUpdateTags(ctx, infra)
+		if err != nil {
+			klog.Errorf("Error processing PVs for infrastructure update: %v", err)
+			return err
+		}
+	}
+	return nil
+}
+
+// fetchPVsAndUpdateTags retrieves all PVs and updates the AWS EFS Access Points tags in batches of 100
+func (c *EFSAccessPointTagsController) fetchPVsAndUpdateTags(ctx context.Context, infra *configv1.Infrastructure) error {
+	pvs, err := c.listPersistentVolumes()
+	if err != nil {
+		return fmt.Errorf("error fetching PVs: %v", err)
+	}
+	// Compute the hash for the new set of tags
+	newTagsHash := computeTagsHash(infra.Status.PlatformStatus.AWS.ResourceTags)
+	pvsToBeUpdated := filterUpdatableVolumes(pvs, newTagsHash)
+
+	// If there are no volumes to update, return early
+	if len(pvsToBeUpdated) == 0 {
+		klog.Infof("No volumes to update as tag hashes are unchanged")
+		return nil
+	}
+	var infraRegion = ""
+	if infra.Status.PlatformStatus != nil && infra.Status.PlatformStatus.AWS != nil {
+		infraRegion = infra.Status.PlatformStatus.AWS.Region
+	}
+	efsClient, err := c.getEFSClient(ctx, infraRegion)
+	if err != nil {
+		return err
+	}
+
+	limiter := rate.NewLimiter(rate.Limit(awsTagsRequestRateLimit), awsTagsRequestBurstSize)
+
+	for _, volume := range pvsToBeUpdated {
+		err = limiter.Wait(context.Background())
+		if err != nil {
+			klog.Errorf("Error waiting for rate limiter: %v", err)
+			return err
+		}
+
+		err = c.updateEFSAccessPointTags(volume, efsClient, infra.Status.PlatformStatus.AWS.ResourceTags)
+		if err != nil {
+			klog.Errorf("Error updating volume's AccessPoint %s tags: %v", volume.Name, err)
+			c.eventRecorder.Warning("EFSAccessPointTagsUpdateFailed", fmt.Sprintf("Failed to update tags for batch %v: %v", volume.Name, err.Error()))
+			continue
+		}
+		// Set the new tag hash annotation in the PV object
+		updatedPv := setPVTagHash(volume, newTagsHash)
+
+		// Update the PV with the new annotations
+		err = c.updateVolumeWithRetry(ctx, updatedPv)
+		if err != nil {
+			klog.Errorf("Error updating PV annotations for volume %s: %v", volume.Name, err)
+			continue
+		}
+		klog.Infof("Successfully updated PV annotations and access points tags for volume %s", volume.Name)
+	}
+	return nil
+}
+
+// updateEFSTags updates the tags of an AWS EFS Access Points
+func (c *EFSAccessPointTagsController) updateEFSAccessPointTags(pv *v1.PersistentVolume, efsClient *efs.EFS, resourceTags []configv1.AWSResourceTag) error {
+
+	tags := newAndUpdatedTags(resourceTags)
+
+	// Create or update the tags
+	_, err := efsClient.TagResource(&efs.TagResourceInput{
+		ResourceId: aws.String(parseAccessPointID(pv.Spec.CSI.VolumeHandle)),
+		Tags:       tags,
+	})
+	if err != nil {
+		klog.Errorf("Error updating tags for PV %s: %v", pv.Spec.CSI.VolumeHandle, err)
+		return err
+	}
+	return nil
+}
+
+func (c *EFSAccessPointTagsController) updateVolumeWithRetry(ctx context.Context, pv *v1.PersistentVolume) error {
```

Comment:
I see an issue with using the syncContext Queue as it will again call the sync function if anything is added to the queue.

I have new retry worker irrespective of the sync function logic and will solve our retry issues.